### PR TITLE
nixos/nextcloud: suggest poolConfig option

### PR DESCRIPTION
**Pull request by Vskilet**
###### Motivation for this change
I suggest to add options for the poolConfig because on my server, default config is really slow and it's missing `pm.max_requests` 
Do you think it's usefull ?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/web-apps/nextcloud.nix [hunk 1]
```diff
@@ -114,6 +114,21 @@ in {
       '';
     };
 
+    poolConfig = mkOption {
+      type = types.str;
```
Comment (eqyiel) [inline nixos/modules/services/web-apps/nextcloud.nix]: I think this might be a `types.lines`?

**Comment by eqyiel** [comment]
Note that you can use `services.phpfpm.pools.nextcloud.extraConfig` like this: https://github.com/eqyiel/dotfiles/blob/dff548627e65fd06a318de4d38f892768eaec1b0/nix/.config/nixpkgs/nixops/realms/maher.fyi/config/nextcloud.nix#L65-L68

I can see how it might be useful to override the default completely though.

**Comment by Vskilet** [comment]
Hi @eqyiel, thanks to review this pull request ! So I follow your guidelines. Is that ok for you ?

**Comment by eqyiel** [comment]
@Vskilet It looks reasonable to me butI don't have commit rights to this repo, so you'll have to convince someone else 😅

**Review by fpletz — APPROVED** [review]
Cool, thanks! It's better to make the pm options overrideable like this so that the other options don't have to be repeated in this case.

**Comment by fpletz** [comment]
@GrahamcOfBorg test nextcloud

**Pull request merged**